File: lista de exercicios/questão 7/questao.py
class Professor:
    def __init__(self, nome):
        self.nome = nome

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)

    def exibir_professores(self):
        print(f"Professores da escola {self.nome}:")
        for professor in self.professores:
            print(professor.nome)

professor1 = Professor("João")
professor2 = Professor("Maria")

escola1 = Escola("Escola A")
escola2 = Escola("Escola B")

escola1.adicionar_professor(professor1)
escola1.adicionar_professor(professor2)

escola2.adicionar_professor(professor1)

escola1.exibir_professores()
escola2.exibir_professores()

File: lista de exercicios/questão 7/questao.go
package main

import "fmt"

type Professor struct {
    Nome string
}

type Escola struct {
    Nome       string
    Professores []Professor
}

func (e *Escola) AdicionarProfessor(professor Professor) {
    e.Professores = append(e.Professores, professor)
}

func (e Escola) ExibirProfessores() {
    fmt.Printf("Professores da escola %s:\n", e.Nome)
    for _, professor := range e.Professores {
        fmt.Println(professor.Nome)
    }
}

func main() {
    professor1 := Professor{Nome: "João"}
    professor2 := Professor{Nome: "Maria"}

    escola1 := Escola{Nome: "Escola A"}
    escola2 := Escola{Nome: "Escola B"}

    escola1.AdicionarProfessor(professor1)
    escola1.AdicionarProfessor(professor2)

    escola2.AdicionarProfessor(professor1)

    escola1.ExibirProfessores()
    escola2.ExibirProfessores()
}
